scipy/sparse/tests/test_array_api.py

```diff
@@ -140,10 +140,16 @@ def test_matmul(A):
     assert np.all((A @ A.T).todense() == A.dot(A.T).todense())
 
 
-@parametrize_square_sparrays
-def test_pow(B):
-    assert isinstance((B**0), scipy.sparse.sparray), "Expected array, got matrix"
-    assert isinstance((B**2), scipy.sparse.sparray), "Expected array, got matrix"
+@parametrize_sparrays
+def test_power_operator(A):
+    assert isinstance((A**2), scipy.sparse.sparray), "Expected array, got matrix"
+
+    # https://github.com/scipy/scipy/issues/15948
+    npt.assert_equal((A**2).todense(), (A.todense())**2)
+
+    # power of zero is all ones (dense) so helpful msg exception
+    with pytest.raises(NotImplementedError, match="zero power"):
+        A**0
```

Comment:
I think the regex on the error message needs an update, `zero power` isn't used above anymore and I get failures locally on this branch:

```
FAILED scipy/sparse/tests/test_array_api.py::test_power_operator[bsr] - AssertionError: Regex pattern did not match.
FAILED scipy/sparse/tests/test_array_api.py::test_power_operator[coo] - AssertionError: Regex pattern did not match.
FAILED scipy/sparse/tests/test_array_api.py::test_power_operator[csc] - AssertionError: Regex pattern did not match.
FAILED scipy/sparse/tests/test_array_api.py::test_power_operator[csr] - AssertionError: Regex pattern did not match.
FAILED scipy/sparse/tests/test_array_api.py::test_power_operator[dia] - AssertionError: Regex pattern did not match.
FAILED scipy/sparse/tests/test_array_api.py::test_power_operator[dok] - AssertionError: Regex pattern did not match.
FAILED scipy/sparse/tests/test_array_api.py::test_power_operator[lil] - AssertionError: Regex pattern did not match.
```

Should be an easy fix, and maybe flushing the CI again will clean up the other failures.

Reply:
Done -- and most of the failures are cleaned up.